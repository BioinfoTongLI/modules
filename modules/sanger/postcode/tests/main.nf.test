// nf-core modules test bioinfotongli/postcode
nextflow_process {

    name "Test Process POSTCODE"
    script "../main.nf"
    process "POSTCODE"

    tag "modules"
    tag "modules_sanger"
    tag "POSTCODE"

    test("iss - csv") {

        config "./iss.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'iss_test'],
                    file(params.modules_testdata_base_path + 'iss_peak_profile_sample.npy', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'iss_peak_locations_sample.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'iss_codebook.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'dummy.txt', checkIfExists: false),
                    6
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("merfish - csv") {

        config "./merfish.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'merfish_test'],
                    file(params.modules_testdata_base_path + 'merfish_peak_profile_sample.npy', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_peak_locations_sample.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_codebook.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_readouts.csv', checkIfExists: true),
                    9
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'merfish_test'],
                    file(params.modules_testdata_base_path + 'merfish_peak_profile_sample.npy', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_peak_locations_sample.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_codebook.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'merfish_readouts.csv', checkIfExists: true),
                    7
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
