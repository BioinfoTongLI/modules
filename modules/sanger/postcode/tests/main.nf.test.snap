{
    "merfish - csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_test"
                        },
                        "merfish_test_decoded_spots.csv:md5,73bb8a4215143145560d15e0e8be1bc4"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ],
                "decoded_peaks": [
                    [
                        {
                            "id": "merfish_test"
                        },
                        "merfish_test_decoded_spots.csv:md5,73bb8a4215143145560d15e0e8be1bc4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T22:24:05.303435026"
    },
    "iss - csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "iss_test"
                        },
                        "iss_test_decoded_spots.csv:md5,a02adfd231d1e365c3eadb1d27aeb8ee"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ],
                "decoded_peaks": [
                    [
                        {
                            "id": "iss_test"
                        },
                        "iss_test_decoded_spots.csv:md5,a02adfd231d1e365c3eadb1d27aeb8ee"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T22:23:42.014978416"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merfish_test"
                        },
                        "merfish_test_decoded_spots.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ],
                "decoded_peaks": [
                    [
                        {
                            "id": "merfish_test"
                        },
                        "merfish_test_decoded_spots.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,74065acdb730aacba1112d6248309b1c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T22:24:15.897845378"
    }
}